### PR TITLE
fix(loadModule): missing getTenantRootModule

### PR DESCRIPTION
**Pull request by JAdshead**
## Motivation and Context

Is possible that root module is built without `getTenantRootModule`.
This leads to a bug where child module requires externals, root module does not provide but its skipped validation.

## How Has This Been Tested?

Test suite and locally building changes into one-app.

## Types of Changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation (adding or updating documentation)
- [ ] Dependency update

## Checklist:

- [ ] My change requires a change to the documentation and I have updated the documentation accordingly.
- [ ] My changes are in sync with the code style of this project.
- [ ] There aren't any other open Pull Requests for the same issue/update.
- [ ] These changes should be applied to a maintenance branch.
- [ ] This change requires cross browser checks.
- [ ] This change impacts caching for client browsers.
- [ ] This change adds additional environment variable requirements for Holocron users.
- [ ] I have added the Apache 2.0 license header to any new files created.

### CODE REVIEW

File: packages/holocron/__tests__/loadModule.node.spec.js [hunk 1]
```diff
@@ -1391,7 +1439,8 @@ describe('loadModule.node', () => {
         // does not have moduleConfig file.
         mockFetch.mockImplementationOnce(() => Promise.resolve({
           status: 404,
-        }));
+        })
+        );
```
Comment (JAdshead) [inline packages/holocron/__tests__/loadModule.node.spec.js:1443]: this comes from eslint

Comment (PixnBits) [reply packages/holocron/__tests__/loadModule.node.spec.js:1443]: wat

File: packages/holocron/src/loadModule.node.js [hunk 2]
```diff
@@ -153,14 +153,14 @@ const loadModuleFallbackExternals = async (baseUrl, moduleName) => {
 const validateLegacyRequiredExternals = ({
   moduleName,
   holocronModule,
-  providedExternals = {},
+  providedExternals,
```
Comment (PixnBits) [inline packages/holocron/src/loadModule.node.js:156]: what's the need for removing the empty default?

File: packages/holocron/__tests__/loadModule.node.spec.js [hunk 3]
```diff
@@ -1254,14 +1261,53 @@ describe('loadModule.node', () => {
       });
     });
 
+    it('does not load child module when root module does not set getTenantRootModule', async () => {
```
Comment (PixnBits) [inline packages/holocron/__tests__/loadModule.node.spec.js:1264]:
```suggestion
    it('does not load child module when root module does not set getTenantRootModule', async () => {
      expect.assertions(1);
```

File: packages/holocron/src/loadModule.node.js (hunk 2 above)
Comment (JAdshead) [reply packages/holocron/src/loadModule.node.js:156]: with the added check on 163 its unnecessary and the default is never used.

**Review by PixnBits — CHANGES_REQUESTED** [review]
I might need to open a PR to add some `expect.assertion` calls to the other `async` tests (out of scope for this PR)
I wonder if we could use an ESLint rule for that? 🤔

File: packages/holocron/src/loadModule.node.js (hunk 2 above)
Comment (PixnBits) [reply packages/holocron/src/loadModule.node.js:156]: huh, I thought 163 was driven from this, rather that the other way around
why is line 163 changed?

**Comment by JAdshead** [comment]
> I might need to open a PR to add some `expect.assertion` calls to the other `async` tests I wonder if we could use an ESLint rule for that? 🤔

Why, it does not provide value for `async` tests. all it does is clog up the failures

File: packages/holocron/src/loadModule.node.js (hunk 2 above)
Comment (PixnBits) [reply packages/holocron/src/loadModule.node.js:156]: Oh! `providedExternals = {}` is not applied when the value to the argument `providedExternals` is something other than `undefined`
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Functions/Default_parameters#passing_undefined_vs._other_falsy_values
Thanks for the reminder 😅

**Comment by PixnBits** [comment]
> Why, [`expect.assertions`] does not provide value for `async` tests. all it does is clog up the failures

I've seen test suites pass but upon inspection they were not returning or awaiting the assertions which, when adjusted, were failing.

FWIW looks like this is already a rule:
https://github.com/jest-community/eslint-plugin-jest/blob/main/docs/rules/prefer-expect-assertions.md

**Comment by JAdshead** [comment]
> > Why, [`expect.assertions`] does not provide value for `async` tests. all it does is clog up the failures
> 
> I've seen test suites pass but upon inspection they were not returning or awaiting the assertions which, when adjusted, were failing.
> 
> FWIW looks like this is already a rule: https://github.com/jest-community/eslint-plugin-jest/blob/main/docs/rules/prefer-expect-assertions.md

this is using `await`, can you show how can the expect not be called without rewriting the tests ?

**Review by Matthew-Mallimo — APPROVED** [review]

**Review by PixnBits — APPROVED** [review]
enabling `jest/prefer-expect-assertions` is a separate scope, so different PR

**Pull request merged**